app/src/main/javascript/bundles/experiment-page/src/TSnePlotViewRoute.js

```diff
@@ -244,6 +255,96 @@ class TSnePlotViewRoute extends React.Component {
     const basename = URI(`experiments/${experimentAccession}${match.path}`, URI(atlasUrl).path()).toString()
 
     const sideTabStyle = {overflow: `clip`, textOverflow: `ellipsis`}
+
+    // Helper function to check if an array is empty and log message
+    const isEmptyArrayAndLog = (route, array, arrayName) => {
+      if (!Array.isArray(array) || array.length === 0) {
+        console.log(`${route.title}: ${arrayName} array is empty`);
+        return true;
+      }
+      return false;
+    };
+
+    var checkIfAnnDataExperiment = function checkIfAnnDataExperiment(experimentAccession) {
```

Comment:
You can rename it as `annDataExperiment`, then the conditional statement reads like this:
```
if (!annDataExperiment(experimentAccession)...
```
Looks much nicer and readable like a sentence.